src/sage/rings/lazy_species.py

```diff
@@ -0,0 +1,1663 @@
+            return P(self[0])
+
+        # f is a constant polynomial
+        if (isinstance(self._coeff_stream, Stream_exact)
+            and not self._coeff_stream._constant
+            and self._coeff_stream._degree == 1):
+            c = self._coeff_stream[0]
+            B = c.parent()
+            if B is ZZ or B is QQ or B == self.base_ring():
+                return P(c)
+            return P(c.coefficients()[0])
+
+        return CompositionSpeciesElement(self, *args)
+
+    def revert(self):
+        r"""
+        Return the compositional inverse of ``self``.
+
+        EXAMPLES::
+
+            sage: L.<X> = LazySpecies(QQ)
+            sage: E1 = L.Sets().restrict(1)
+            sage: g = E1.revert()
+            sage: g[:5]
+            [X, -E_2, -E_3 + X*E_2, -E_4 + E_2(E_2) + X*E_3 - X^2*E_2]
+
+            sage: E = L.Sets()
+            sage: P = E(X*E1(-X))*(1+X) - 1
+            sage: P.revert()[:5]
+            [X, X^2, X*E_2 + 2*X^3, X*E_3 + 2*X^2*E_2 + E_2(X^2) + 5*X^4]
+
+        TESTS::
+
+            sage: (3 + 2*X).revert()
+            (-3/2) + 1/2*X
+        """
+        P = self.parent()
+        if P._arity != 1:
+            raise ValueError("arity must be equal to 1")
+        coeff_stream = self._coeff_stream
+        if isinstance(coeff_stream, Stream_zero):
+            raise ValueError("compositional inverse does not exist")
+        R = P._laurent_poly_ring
+        if (isinstance(coeff_stream, Stream_exact)
+            and coeff_stream.order() >= 0
+            and coeff_stream._degree == 2):
+            # self = a + b * X; self.revert() = -a/b + 1/b * X
+            a = coeff_stream[0]
+            b = coeff_stream[1].coefficients()[0]
+            X = R(SymmetricGroup(1))  # as a polynomial species
+            coeff_stream = Stream_exact((-a/b, 1/b * X),
+                                        order=0)
+            return P.element_class(P, coeff_stream)
+
+        # TODO: coefficients should not be checked here, it prevents
+        # us from using self.define in some cases!
+        if coeff_stream[0]:
+            raise ValueError("cannot determine whether the compositional inverse exists")
+
+        X_mol = P._laurent_poly_ring._indices.subset(1)[0]  # as a molecular species
+        X = P(SymmetricGroup(1))  # as a lazy species
+
+        def coefficient(n):
+            if n:
+                return 0
+            c = coeff_stream[1].coefficient(X_mol)
+            if c.is_unit():
+                return ~c
+            raise ValueError("compositional inverse does not exist")
+
+        b = P(lambda n: 0 if n else coeff_stream[1].coefficient(X_mol))  # TODO: we want a lazy version of Stream_exact
+        b_inv = P(coefficient)  # TODO: we want a lazy version of Stream_exact
+        g = P.undefined(valuation=1)
+        g.define(b_inv * (X - (self - b * X)(g)))
+        return g
+
+    compositional_inverse = revert
+
+
+class LazySpeciesElement_generating_series_mixin:
```

Comment:
```suggestion
class GeneratingSeriesMixin:
```
I think a more concise name is better here. The docstring does the link with `LazySpeciesElement`. If you want the long name, then let’s follow Python naming conventions.

Reply:
I used the idea from `LazyPowerSeries_gcd_mixin`.  But yes, it seems that camel case is more conventional:
```
combinat/diagram_algebras.py:2162:class UnitDiagramMixin:
combinat/partition_kleshchev.py:863:class KleshchevCrystalMixin:
combinat/crystals/tensor_product.py:656:class QueerSuperCrystalsMixin:
groups/galois_group.py:48:class _GMixin:
groups/perm_gps/symgp_conjugacy_class.py:18:class SymmetricGroupConjugacyClassMixin:
groups/libgap_mixin.py:23:class GroupMixinLibGAP:
modules/fp_graded/steenrod/module.py:102:class SteenrodModuleMixin:
rings/lazy_species.py:799:class LazySpeciesElement_generating_series_mixin:
rings/lazy_series.py:6221:class LazyPowerSeries_gcd_mixin:
```
It seems to me that the main class the mixin is intended for is usually part of the name, so I'd go with `LazySpeciesElementGeneratingSeriesMixin`, if that's ok for you?

Reply:
If I named that other class, I will take the full blame. Otherwise I missed that, and I take full blame. The CamelCase would have been best for that class too.